feat: add support to luxuretv.com

Barbarella6666666 opened the pull request.

Barbarella6666666 committed:
feat: add support to luxuretv.com

Barbarella6666666 committed:
refactor: fix search title

Barbarella6666666 committed:
ruff: fixing code format

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,75 @@
+    @error_handling_wrapper
+    async def search(self, scrape_item: ScrapeItem) -> None:
+        title: str = ""
+        url = scrape_item.url.with_path(scrape_item.url.path + "/")
```
You probably need a trailing slash for all pages.

Set `DEFAULT_TRIM_URLS` in the class to `False

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,75 @@
+    @error_handling_wrapper
+    async def search(self, scrape_item: ScrapeItem) -> None:
+        title: str = ""
+        url = scrape_item.url.with_path(scrape_item.url.path + "/")
```
They have bad routes handling. The trailing slash is only needed for the first page and returns 404 for any other page

Set `DEFAULT_TRIM_URLS` in the class to `False` and you should also can add a check to only add it to the first page

Barbarella6666666 replied:
The trailing slash is only added to the first page already. I've set DEFAULT_TRIM_URLS to False, but it is only used in parse_url method, not in the first page. Let me know if I misunderstood anything.

NTFSvolume replied:
Only add it if the URL does not end in `.html`
```suggestion
        url = scrape_item.url
        if url.name and not url.name.endswith(".html"):
            url = url / ""
```

NTFSvolume replied:
This is to handle the case where the input URL from the user was not the first page

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,75 @@
+
+
+class LuxureTVCrawler(Crawler):
+    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {"Video": "/videos/...", "Search": "/searchgate/videos/<search>/..."}
```
```suggestion
    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {"Video": "/videos/<name>-<id>.html", "Search": "/searchgate/videos/<search>/...",}
```

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,75 @@
+
+class Selector:
+    VIDEO_PLAYER = "video.video-js > source"
+    METADATA_SELECTOR = "script[type='application/ld+json']"
```
This one is not used
```suggestion
```

Barbarella6666666 committed:
refactor: Applying requested changes

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,78 @@
+class Selector:
+    VIDEO_PLAYER = "video.video-js > source"
+    TITLE = "h1.title-right"
+    NEXT_PAGE = "div#pagination > a:-soup-contains('Suivante »')"
```
They also have english and spanish versions of the site: en.luxuretv, es.luxuretv

I think changing it to just this will still work:
```suggestion
    NEXT_PAGE = "div#pagination > a:-soup-contains('»')"
```

Barbarella6666666 committed:
refactor: applying suggested changes

NTFSvolume commented on cyberdrop_dl/crawlers/luxuretv.py:
```diff
@@ -0,0 +1,80 @@
+                title = css.select_one_get_text(soup, Selector.TITLE)
+                title = title.split(":")[-1].strip()
+                title = self.create_title(f"{title} - [search]")
+                scrape_item.setup_as_album(title)
```
We can use the query bypass that and also skip the selector as well

```suggestion
    async def search(self, scrape_item: ScrapeItem, query: str) -> None:
        title = query.replace(".", " ").capitalize()
        title = self.create_title(f"{title} [search]")
        scrape_item.setup_as_album(title)
        url = scrape_item.url
        if url.name and not url.name.endswith(".html"):
            url = url / ""
            
        async for soup in self.web_pager(url, cffi=True):        
```

NTFSvolume replied:
This is grabbing ` - Page X` with the title itself when the input is not the first page

NTFSvolume replied:
This is grabbing ` - Page X` with the title itself when the input is not the first page

NTFSvolume committed:
fix: do not include page number in title

Title had ` - Page X` in it when the input is not the first page. Use query instead
